### DISCUSSION TITLE
setting up Workgroups

### DISCUSSION THREAD

**Discussion by blackcrack** [snappysnappydog/Simple-IP-Config#121] [Ideas]
Hi,

long time is ago.. hey, nice to see you there on github !! :)))))

and hey, can you add a Setting up the Workgroup  ?

and by the way, i have W764bit on run, there works not really well ..
but reactos is comming ! :)

but the best and further like this, keep it up :) :+1: 

best regards
Blacky

**Comment by blackcrack** [comment]
jing.. and bring to top... could anyone help to add a code to setting up the workgroup ?
btw, @KurtisLiggett , you're alive ?
best

**Comment by blackcrack** [comment]
ping...

**Comment by KurtisLiggett** [comment]
I probably will not be implementing this, but anyone else is welcome to go for it.
Doesn't Windows need a restart after changing the Workgroup?

**Reply by blackcrack** [reply]
not really, i guess it was a peculiarity of windows or the programmers where have some implement, if the workgroup change, then become a other direction.. and not the whole driver-stack have change, easy the affiliation belonging or Membership of a group.. an Linux with the network stack don't need also to have a boot, because it's not a thing of System or driver.. I guess, at Windows 10/11 it is this boot no more need.
And by side, other Network Configuration programs have include this workgroup-changer include for a long time..

**Comment by blackcrack** [comment]
@KurtisLiggett 

check it out:
"net config"

```
i=XXXXXXX
For /F "tokens=3" %%i in ('Net Config "Workstation^|Find /i "Workstation Domain") Do @set WORKGROUP=%%i&SETX
WORKGROUP=%%i
```
the setx is out from the resourcekit, this can you add to your simple-ip-config .
or "netdom.exe" ... or you add the code from netdom to your proggy..
it is from the recoucekid of NT4.0 :  
[http://www.procompsys.de/Hilfe/katalog/netdom_v1.8_x86_aus_ResKit_NT4.zip](http://www.procompsys.de/Hilfe/katalog/netdom_v1.8_x86_aus_ResKit_NT4.zip)
test it out and if works in all WinNT-Versions have you won.

there is the computername
```
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\ComputerName
```

Command : "net" /?

net VIEW

therewith can you the Netgroupnames in the net.. imho

and check it out :  
wmic.exe therewith do you become many out of the net and configuration.

```
@echo off
 start /B /W wmic.exe /interactive:off ComputerSystem Where "Name='%computername%'" Call UnJoinDomainOrWorkgroup FUnjoinOptions=0
 start /B /W wmic.exe /interactive:off ComputerSystem Where "Name='%computername%'" Call JoinDomainOrWorkgroup name="WORKGROUP"
 start /B /W wmic.exe /interactive:off ComputerSystem Where "Name='%computername%'" Call Rename name="standaloneComputer"
 shutdown.exe /r /t 0 /f
```
shutdown/reboot must not be.. have fount it at a [webpage](https://www.mcseboard.de/topic/208824-per-script-einen-rechner-aus-der-dom%C3%A4ne-nehmen-und-in-eine-arbeitsgruppe-eintragen/) where play around with them..


best

**Reply by blackcrack** [reply]
@KurtisLiggett i have font a 2nd webpage who describes the whole at Winaero :   
[Change Workgroup Name in Command Prompt](https://winaero.com/change-workgroup-name-windows-10/#Change_Workgroup_Name_in_Command_Prompt)  
  
wmic computersystem where name="%computername%" call joindomainorworkgroup name="Workgroup_Name"  
  
then read out the Workgroup :  
[How to find information about a computer in a Workgroup](http://kb.mit.edu/confluence/display/mitcontrib/How+to+find+information+about+a+computer+in+a+Workgroup)  
  
nbtstat -a %computername%  
  
  
therewith have we a possible for read out the actually workgroupname, this could be grab  
and push it to a Workgroup:[\_\_\_textfild_____] and is it not in a workgroup, is this fild free, because grab"0"
because 2 parts is nothing or the line with "Groups" exist easy not.

the Textfild could make as variable $systemworkgroup and add to the line 

wmic computersystem where name="%computername%" call joindomainorworkgroup name="$systemworkgroup"

= 
Simple IP Config starts and execute 
"nbtstat -a %computername%  "
graps the group name and set it up as $systemworkgroup and set it as Default Textfild entry and show whitch Groupname the computer have.  
If no group name seted behind Install is the groupentry "" so, not setted up. and therewith is the default systemworkgroup ""
and "" is pusched in the textfild=systemworkgroup  
  
is a group seted, can be changed via   
wmic computersystem where name="%computername%" call joindomainorworkgroup name="$systemworkgroup"  
  
[Remove the Workgroup connectivity] $systemworkgroup=""
wmic computersystem where name="%computername%" call joindomainorworkgroup name="$systemworkgroup"

therewith is the Systemworkgroup removed,  
can be changed .. and can be set up 

@KurtisLiggett  would e very nice if you could add it :)
best

**Comment by KurtisLiggett** [comment]
@blackcrack Any technique is going to require a system restart. All methods end with the last step "restart your computer".
To clarify, are you requesting to make the change but allow the user to restart later? This is possible.
Or are you wanting to change the workgroup and have the changes applied immediately without rebooting? This is not possible.

Also, does the workgroup not already show up in the lower right corner? Note there is an issue #166 for the description.

In any case, I've added workgroups to the [project roadmap](https://github.com/KurtisLiggett/Simple-IP-Config/discussions/165) for the next release. We'll see what happens!

**Comment by blackcrack** [comment]
restart later, yes.. so why not.. in any case, the ip's and the Workgroup is possible to setting up..
the System is going in anyway to a reboot.. early or later.. so..

i have test it by self at my workstation here, where i write, there is not coming a "you have do a restart"
the commandline was a silence change and i have not become a "restart your computer" at cmd-line
maybe a Set like "Please reboot" in red letters if the Name of workgroup if has change, the user can do it later..
(or you add a Admin-dependent Stop/start Lanman-service Button if is the WinNT-User in the Admin-Group (Owner),  
or the User have start the Simple-IP-config as Admin)    
  
If i change the Workgroup, do i must not reboot, i can explore the //Linuxserver/Windows directly because the name and passwort at my computers be the same, on the server from Linux (so the user-smb-passwd) and the UserName, because this can i change directly the Workgroupname in WinNT, like WinXP, Win7 and Win10 and can directly browse in the //Linuxserver/~  ..  
So, now my question, why think the peoples ever, they must first reboot before the Workgroups name overtaken?
Because ever comes from the Microsoft ordering System the "Warning" to have boot befor you can use.. this is absolutly wrong imho, 
because i see it at my self Computers.. it works still ! i can rename the Workgroup name and it's comes a "Welcome in your Workgroup"
or something..  so, you can check it out by self, if you play around with the Workgroupname ... Only because the System say "you have to boot" must noch lange nicht booten, to be comes a connecting connection to the smb://  .. in linux you don't have to boot either ..
this is only a have to make a special option.. and this makes me angry by Microsoft.. like "This is a Serversystem" This is more Expensive..
but this is bullschit and only a regestrystring tell to be a Serversystem because it's a MS-Own HTTP-Webserver and a FTP and a Mailserver at this "Serversystem" running.. with multiuser and some more.. but what is a workstaions, nothing else, without Server-Programms who service different Services like Web, FTP/sftp/ssh and a Mailserver  a workstation (and without them reg-string) costs for a Private Home 100$, a WinNT Pro 100 and a Couples more and a Server directly 10x or 100x at $ more which is pure window dressing, but with a Homeversion can you build up also an Web,sftp and other Servers with foreigner Service Programs , so what should the Joke .. imho..

and this with "you have to boot" is the same.. it makes more "special" , i play since 25 years around with Computers..
Dos,Win95,Suse4.5,Redhat short, then long time Mandrake/Mandriva by side of WinNT4 with IE4, 2000, XP up to Win7 and Fedora now..

i be very happy to be now (hopefully) in the next release and wish you a happy play around with the networkgroups :)
i guess you want see, if you make it via cmd and your program, do you become no Dialog for reboot.. it's need's imho also not,
because, is the workgroup selected or setting up, is the computer still in the workgroup and do his work.. imho..
and it is easy a thing of the "SMB-Stack" and not a thing of the System.. and there do you need only restart the Lanman-service, that's all

so,  i am really curios how and when it want comes and happens :)))
a big fad thank's in front @KurtisLiggett :)

best :)

**Comment by KurtisLiggett** [comment]
@blackcrack would you mind trying this test version? 
In the _Tools_ menu, there is an entry for "Change Workgroup". Try it out. Is that what you were expecting?
[Simple IP Config 2.9.7 - workgroup_test.zip](https://github.com/KurtisLiggett/Simple-IP-Config/files/9499868/Simple.IP.Config.2.9.7.-.workgroup_test.zip)

**Reply by blackcrack** [reply]
@KurtisLiggett  , cool :)

i test it directly out  
 
and by the way, i have found in the net  
this for batch (cmd):  
```
for /f "usebackq skip=1 tokens=*" %%i in (`wmic computersystem get workgroup ^| findstr /r /v "^$"`) do set workgroup=%%i
echo %workgroup%
```


and here the test :  

![Screen 00000](https://user-images.githubusercontent.com/7925717/188871095-564edcfd-0cc9-4ee9-ae24-0ed144c045ba.jpg)

![Screen 00001](https://user-images.githubusercontent.com/7925717/188872665-7b64ee84-db67-4098-a683-e1ee921566f4.jpg)

**Reply by blackcrack** [reply]
now would it grate if the Workgroup config directly unter the DNS-Configuration, because to be inside of the configurations if one selected..

under  
* [ ] Adressen registrieren
---------------------------------------
```
 Member of a Workgroup                             Member of a Domain
[__load actual workgroup_as def. entry_]         [____________________________]
```

loading the actually Workstation and set as default text in Workgroup,
therewith it is see to exist already setted up workstation and able to save in the profiles.ini

hmmmmmmm.......

* [ ] Adressen registrieren
---------------------------------------
```
Name of the Computer                     Member of a Workgroup                             Member of a Domain
[_Computername__________]          [__load actual workgroup_as def. entry_]         [____________________________]

DNS-Suffix       
[_de____________________]  
```

just as example :)

**Comment by KurtisLiggett** [comment]
Did the Simple IP Config test not work? I don't know why you are giving command line alternatives if it is already working. I am using the WMI "Win32_ComputerSystem" class and "JoinDomainOrWorkgroup" method, which should be more reliable than a command alternative.

**Comment by blackcrack** [comment]
n0000, it works :) 
and yes agreed  "JoinDomainOrWorkgroup" method.
but win32, it want support also the x86, hmmm... you should add a selector
of Architecture, x86 and 64Bit, so a checker if the actual Architecture a x86 or an x86_64
and then let call the right architecture.. ... as suggestion therewith 
do you serve in future both Architectures.. if you looking on my batchfile,
there can i set up %arch% and it will tell x64 or x86 like at the MS-Standart
setup files, i can change the whole files with add %arch% and 
a subfolder arch for special files who have no %arch% in the name..

but was easy meant as alternative to check out the workgroup,
the alternate the "wmic computersystem get workgroup"
to have a more easy version to become the Workgroup Name into a Variable
what i have now use in my admin setup-update batch-file

i have now create a new version of my setup-update-batch file,
you can start it with switches or easy as normal user and as admin..

if you start this batch as admin can you manipulate the workgroup,
check it out if you want:  

--=>> [here can you a bit explore the files](https://www.blackysgate.de/files/)  
  
--=>> [setup-update-batch-1.74.bat%202022-09-07%2018-32-47.zip](https://www.blackysgate.de/files/folder-root/batch-files/setup-update-batch-1.74.bat%202022-09-07%2018-32-47.zip)  
  
the workgroup tweaker is under 1, if you admin ;)

best and manny fun :)
